fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountTransaction.java

```diff
@@ -345,6 +348,7 @@ private SavingsAccountTransaction(final SavingsAccount savingsAccount, final Off
         this.reversed = isReversed;
         this.paymentDetail = paymentDetail;
         this.createdDate = createdDate;
+        this.submittedOnDate = DateUtils.getBusinessLocalDate();
```

Comment:
There are multiple constructor of the savings transaction. It might worth to cover all places when a transaction can be created, the submitted on date is set as well!